### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Timi2424**
[DEMO LINK](https://Timi2424.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,62 @@
...
+      break;
+
+    case Status.Completed:
+      selectedTodos = todos.filter(currentTodo => currentTodo.completed);
+      break;
+
+    default:
+      selectedTodos = todos;
+      break;
+  }
```
Comment (mpiskadlo) [inline src/App.tsx:46]: I would go for something like this:
```suggestion
  const filteredTodos = () => {
    switch (filterByStatus) {
    case Status.Active:
      return todos.filter(currentTodo => !currentTodo.completed);

    case Status.Completed:
      return todos.filter(currentTodo => currentTodo.completed);

    default:
      return todos;
  }
  };
```

And similar thing for `visibleTodos` ;)

Comment (mpiskadlo) [reply src/App.tsx:46]: And then in JSX you just do `filteredTodos().map`

**Review by mpiskadlo — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Timi2424) [reply src/App.tsx:46]: what an oversight, thanks!

Comment (Timi2424) [reply src/App.tsx:46]: I don't know exactly why do you want me to map it. I hope my refactor is sufficient

**Review by Barosz30 — APPROVED** [review]
Seems just right :)

**Review by mbruhler — APPROVED** [review]
Well done!

**Pull request closed without merging**